addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -24,13 +32,25 @@ const GalleryWidget = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     _onClickImg: function (ev) {
-        if (this.$modal || ev.currentTarget.matches("a > img")) {
+        const clickedEl = ev.currentTarget;
+        if (this.$modal || clickedEl.matches("a > img")) {
             return;
         }
         var self = this;
-        var $cur = $(ev.currentTarget);
 
-        var $images = $cur.closest('.s_image_gallery').find('img');
+        let imageEls = this.target.querySelectorAll("img");
+        const currentImageEl = clickedEl.closest("img");
+        const currentImageIndex = [...imageEls].indexOf(currentImageEl);
+        // We need to reset the images to their original source because it might
+        // have been changed by a mouse event (e.g. "hover effect" animation).
+        let index = 0;
+        imageEls = [...imageEls].map(el => {
+            const cloneEl = el.cloneNode(true);
+            cloneEl.src = this.originalSources[index];
+            index++;
+            return cloneEl;
+        });
```

Comment:
Nitpick:
It could be confusing to have `index` and `currentImageIndex`.

It works but IMO it's better like this
```js
for (let i = 0; i < imageEls.length; i++) {
  const cloneEl = imageEls[i].cloneNode(true);
  cloneEl.src = this.originalSources[i];
  imageEls[i] = cloneEl;
}
```
but again, it's just a matter of taste. You can keep your version if you prefer :slightly_smiling_face:

Reply:
I replaced "index" by "i" so it will be clearer I think